core/src/commonMain/kotlin/komem/litmus/LitmusAutoState.kt

```diff
@@ -0,0 +1,72 @@
+package komem.litmus
+
+/**
+ * A convenience interface to simplify specifying outcomes.
+ *
+ * All classes implementing this interface provide some r1, r2, ... variables
+ * to write the outcome into. If a litmus test's state extends one of these classes,
+ * specifying `outcome { ... }` is not necessary, as it will be inferred from r1, r2, ...
+ *
+ * These classes are also used as outcomes themselves to better utilize resources.
+ */
+interface LitmusAutoState {
+    override fun toString(): String
+    override fun hashCode(): Int
```

Comment:
why you added these overrides here?
there have no effect

Reply:
They do have an effect:

* `toString()` is used to display the outcome in the table, and if not overridden, it would also display the other variables of a particular state
* `hashCode()` is used in the fallback case (which is any non-user-specified outcome) to put it into a map

Reply:
interface makes these functions abstract and forces to implement them in subclasses, agreed to leave a comment